MoreCyclopsUpgrades/Monobehaviors/CyBioReactorMono.cs

```diff
@@ -0,0 +1,446 @@
+
+                foreach (BioEnergy material in materials)
+                {
+                    InventoryItem inventoryItem = this.Container.AddItem(material.Pickupable);
+                    this.MaterialsProcessing.Add(material);
+                    material.Size = inventoryItem.width * inventoryItem.height;
+                }
+            }
+
+            isLoadingSaveData = false;
+        }
+
+        public void ConnectToCyclops(SubRoot parentCyclops)
+        {
+            QuickLogger.Debug("BioReactor has been connected to Cyclops", true);
+
+            this.ParentCyclops = parentCyclops;
+
+            this.transform.SetParent(parentCyclops.transform);
+        }
+
+        public void ConnectToInventory(Dictionary<InventoryItem, uGUI_ItemIcon> lookup)
+        {
+            if (this.InventoryMapping is null)
+            {
+                this.InventoryMapping = lookup;
+
+                this.Container.onAddItem += OnAddItemLate;
+
+                if (this.MaterialsProcessing.Count == 0)
+                    return;
+
+                foreach (KeyValuePair<InventoryItem, uGUI_ItemIcon> pair in lookup)
+                {
+                    InventoryItem item = pair.Key;
+                    uGUI_ItemIcon icon = pair.Value;
+
+                    BioEnergy bioEnergy = this.MaterialsProcessing.Find(m => m.Pickupable == item.item);
+
+                    if (bioEnergy is null)
+                    {
+                        QuickLogger.Debug("Matching pickable in bioreactor not found", true);
+                        continue;
+                    }
+
+                    AddDisplayText(bioEnergy, icon);
+                }
+            }
+        }
+
+        private void AddDisplayText(BioEnergy bioEnergy, uGUI_ItemIcon icon)
+        {
+            var arial = (Font)Resources.GetBuiltinResource(typeof(Font), "Arial.ttf");
+
+            var textGO = new GameObject("EnergyLabel");
+
+            textGO.transform.parent = icon.transform;
+            textGO.AddComponent<Text>();
+
+            Text text = textGO.GetComponent<Text>();
+            text.font = arial;
+            text.material = arial.material;
+            text.text = string.Empty;
+            text.fontSize = 13 + bioEnergy.Size;
+            text.alignment = TextAnchor.MiddleCenter;
+            text.color = Color.yellow;
+
+            Outline outline = textGO.AddComponent<Outline>();
+            outline.effectColor = Color.black;
+
+            RectTransform rectTransform = text.GetComponent<RectTransform>();
+            rectTransform.localScale = Vector3.one;
+            rectTransform.anchoredPosition3D = Vector3.zero;
+
+            bioEnergy.DisplayText = text;
+        }
+
+        [ProtoMember(3)]
+        [NonSerialized]
+        public float _constructed = 1f;
+
+        [AssertNotNull]
+        public ChildObjectIdentifier storageRoot;
+
+        internal List<BioEnergy> MaterialsProcessing { get; } = new List<BioEnergy>();
```

Comment:
Move lists to top of file. Initialize capacity on `MaterialsProcessing` even though it will still comes out to be the default value.